test/recur_test.js

```diff
@@ -26,56 +26,56 @@ suite('recur', function() {
       });
     }
 
-    function checkThrow(data, expectedMessage, dtstart, stack) {
+    function checkNoInstance(data, expectedMessage, dtstart, stack) {
       test(expectedMessage, function() {
         let recur = new ICAL.Recur(data);
         if (dtstart) {
           dtstart = ICAL.Time.fromString(dtstart);
         } else {
           dtstart = ICAL.Time.epochTime.clone();
         }
-        assert.throws(function() {
-          recur.iterator(dtstart);
-        }, expectedMessage);
+
+        const iter = recur.iterator(dtstart);
+        assert.equal(iter.next(), null);
       });
     }
 
-    checkThrow({
+    checkNoInstance({
       parts: {
         BYYEARDAY: [3, 4, 5],
         BYMONTH: [2]
       }
     }, 'Invalid BYYEARDAY rule');
 
-    checkThrow({
+    checkNoInstance({
       parts: {
         BYWEEKNO: [3],
         BYMONTHDAY: [2]
       }
    }, 'BYWEEKNO does not fit to BYMONTHDAY');
 
-    checkThrow({
+    checkNoInstance({
       freq: 'MONTHLY',
       parts: {
         BYWEEKNO: [30]
       }
     }, 'For MONTHLY recurrences neither BYYEARDAY nor BYWEEKNO may appear');
 
-    checkThrow({
+    checkNoInstance({
       freq: 'WEEKLY',
       parts: {
         BYMONTHDAY: [20]
       }
     }, 'For WEEKLY recurrences neither BYMONTHDAY nor BYYEARDAY may appear');
 
-    checkThrow({
+    checkNoInstance({
       freq: 'DAILY',
       parts: {
         BYYEARDAY: [200]
       }
     }, 'BYYEARDAY may only appear in YEARLY rules');
```

Comment:
The robustness principle suggests to me that we should only error if we reach a point where we simply can't interpret the file any longer. I'd be okay with using `strict` as a guard here if that makes sense to you. Probably the _best_ thing to do in this instance is to favor the `FREQ` property and any invalid parts that follow are silently dropped, resulting in occurrences that follow whatever can be validly interpreted. Unfortunately, the spec provides rules for how to _create_ valid iCalendar objects, but no real guidance on how to deal with invalid ones.

Reply:
Yeah, generally I'd like ical.js to be able to parse everything and be lenient. The strict mode was meant for the case where you might use ical.js as a validator and would want it to bail. Let's put this behind the strict mode check.

Reply:
I wasn't quite sure how to access whether we were in strict mode from the recurrence iter, so I've implemented the infinite loop prevention without modifying any throwing behavior. If that behavior is desire, it may be best to address that in a followup.